tests/system/test_agent_keep_alive/test_agent_keep_alive.py

```diff
@@ -0,0 +1,339 @@
+            host_manager.run_command('wazuh-manager', f"ip addr add {network['manager_network'][1]} dev eth0")
+            host_manager.run_command('wazuh-manager', f"ip addr add {network['manager_network'][2]} dev eth0")
+
+        # Restore agent network configuration
+        if 'ipv6' in configuration['agent_network']:
+            host_manager.run_command('wazuh-agent1', f"ip addr add {network['agent_network'][0]} dev eth0")
+            host_manager.run_command('wazuh-agent1', 'ip route add 172.24.27.0/24 via 0.0.0.0 dev eth0')
+        elif 'ipv4' in configuration['agent_network']:
+            host_manager.run_command('wazuh-agent1', f"ip addr add {network['agent_network'][1]} dev eth0")
+            host_manager.run_command('wazuh-agent1', f"ip addr add {network['agent_network'][2]} dev eth0")
+
+
+@pytest.fixture(scope='function')
+def modify_ip_address_conf(test_case):
+
+    with open(agent_conf_file, 'r') as file:
+        old_agent_configuration = file.read()
+
+    with open(manager_conf_file, 'r') as file:
+        old_manager_configuration = file.read()
+
+    for configuration in test_case['test_case']:
+        if 'yes' in configuration['ipv6_enabled']:
+            new_manager_configuration = old_manager_configuration.replace('IPV6_ENABLED', "'yes'")
+        else:
+            new_manager_configuration = old_manager_configuration.replace('IPV6_ENABLED', "'no'")
+
+        if 'ipv4' in configuration['ip_type']:
+            new_configuration = old_agent_configuration.replace('<address>MANAGER_IP</address>',
+                                                                f"<address>{network['manager_network'][0]}</address>")
+            host_manager.modify_file_content(host='wazuh-agent1', path='/var/ossec/etc/ossec.conf',
+                                             content=new_configuration)
+        elif 'ipv6' in configuration['ip_type']:
+            new_configuration = old_agent_configuration.replace('<address>MANAGER_IP</address>',
+                                                                f"<address>{network['manager_network'][1]}</address>")
+            host_manager.modify_file_content(host='wazuh-agent1', path='/var/ossec/etc/ossec.conf',
+                                             content=new_configuration)
+        elif 'dns' in configuration['ip_type']:
+            new_configuration = old_agent_configuration.replace('<address>MANAGER_IP</address>',
+                                                                '<address>wazuh-manager</address>')
+            host_manager.modify_file_content(host='wazuh-agent1', path='/var/ossec/etc/ossec.conf',
+                                             content=new_configuration)
+
+    with open(manager_conf_file, 'w') as file:
+        file.write(new_manager_configuration)
+
+    host_manager.apply_config(manager_conf_file)
+
+    yield
+
+    with open(manager_conf_file, 'w') as file:
+        file.write(old_manager_configuration)
```

Comment:
Please refactor this function to avoid unnecessary code repetition.

Reply:
Done in commit a765962